app/services/ui_notifications/hosts/provision.rb

```diff
@@ -0,0 +1,43 @@
+module UINotifications
+  module Hosts
+    class Provision < Base
+      private
+      def build_completed
+        blueprint = NotificationBlueprint.find_by(name: 'host_build_completed')
+
+        if blueprint.notifications.where(:subject => subject).empty?
+          logger.debug('creating new UI notification host_build_completed')
+          ::Notification.create!(
+            initiator: initiator,
+            subject: subject,
+            audience: audience,
+            notification_blueprint: blueprint
+            )
+        else
+          logger.debug('UI notification already exists, updating timestamps.')
+          blueprint.notifications.
+            where(subject: subject).
+            update_all(expired_at: blueprint.expired_at)
+        end
+      end
+
+      def destroyed
+        # IMHO no point in keeping notifications for deleted hosts, as it may
+        # contain links to non existing actions.
+        logger.debug("Removing all notifications for host: #{subject}")
+        Notification.where(subject: subject).delete_all
```

Comment:
`.destroy_all`, if I'm not wrong `delete_all` is just going to delete stuff in the Notifications table (not NotificationRecipients)